source/ur_api.cpp

```diff
@@ -2472,6 +2502,62 @@ urUSMGetMemAllocInfo(
     return result;
 }
 
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Create USM memory pool with desired properties.
+///
+/// @details
+///     - UR can create multiple instances of the pool depending on allocation
```

Comment:
@igchor : what do we mean by multiple instances? let's say we have a pool of 1GB requested, and we have two device allocations with different flags.

Does that mean UR could create 2 instances of the same size, each with a different flag to support the two device allocations requested.

Or does it mean UR could create two native memory locations, each with the size of the device allocation requested, and associate them both as belonging to the same pool?

Reply:
Right now, I would say it's quite open for interpretation, but what I had in mind for L0 was the second option you mentioned (letting UR create 'sub-pools' for different allocation flags). At least for different flags, there would still be multiple 1GB instances of the pool for multiple devices in that case.

In general, I would like to work (in the next PR) on extending ur_usm_pool_limits_desc_t with different/more specific options so that it would be clear how much memory can be used for a memory pool (perhaps even on different levels - per context, per device, etc.)

Reply:
thanks @igchor . Second option, where UR adapter can create sub-pools aligns with what I would expect. Thanks!